### PR TITLE
[MNG-5180] - Versioning's snapshot version list is not included in metadata merge 

### PR DESCRIPTION
**Pull request by fuss86**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by fuss86** [comment]
Based on http://maven.apache.org/ref/3.6.0/maven-repository-metadata/repository-metadata.html

> Versioning information for a sub-artifact of the current snapshot artifact.

this fix is wrong

**Comment by michael-o** [comment]
@fuss86 We are currently trying to pick this up. Can you remember why you consider this to be wrong and #371 to be correct? Note that Resolver uses the #371 approach.

File: maven-repository-metadata/src/main/mdo/metadata.mdo [hunk 1]
```diff
@@ -202,6 +202,30 @@ under the License.
...
+
+                for ( SnapshotVersion snapshotVersion : versioning.getSnapshotVersions() )
+                {
+                    boolean exists = false;
+
+                    for ( SnapshotVersion sv : v.getSnapshotVersions() )
+                    {
+                        if ( java.util.Objects.equals( snapshotVersion.getClassifier(), sv.getClassifier() ) &&
+                             java.util.Objects.equals( snapshotVersion.getExtension(), sv.getExtension() ) &&
+                             java.util.Objects.equals( snapshotVersion.getUpdated(), sv.getUpdated() ) &&
```
Comment (kwin) [inline maven-repository-metadata/src/main/mdo/metadata.mdo:214]: This is not correct, as only classifier, extension and version act as key. The updated may actually differ but the should not lead to duplicate keys.

**Comment by jira-importer** [comment]
Resolve #7003

**Pull request closed without merging**